client/common/common.c

```diff
@@ -363,7 +363,7 @@ do_getopt(struct client *client, int *argc, char **argv[])
             {
                 char *ptr;
                 client->lines = strtol(optarg, &ptr, 10);
-                client->lines_mode = (!strcmp(ptr + 1, "up") ? BM_LINES_UP : BM_LINES_DOWN);
+                client->lines_mode = (*ptr && !strcmp(ptr + 1, "up") ? BM_LINES_UP : BM_LINES_DOWN);
```

Comment:
Should this not be `ptr` instead of `*ptr` if the purpose is to check if `ptr` is not NULL?

Reply:
The idea is not to check if `ptr` is not NULL (in fact, `strtol` guarantees that `ptr` is not NULL), but rather to check if `strtol` parsed the entire string, i.e. if `optarg` consists of only digits.

If that's the case then `ptr` will point to the `NUL` terminator, so `*ptr = 0` and after the changes `strcmp(ptr + 1, ...)` will *not* be called.

Previously, in this case it was calling `strcmp(ptr + 1, ...)`, which was an out-of-bounds read because `ptr + 1` is right past the `NUL` terminator.